action-types.yml

```diff
@@ -0,0 +1,133 @@
+# See https://github.com/typesafegithub/github-actions-typing/
+inputs:
+  name:
+    type: string
+  
+  version:
+    type: string
+  
+  url:
+    type: string
+  
+  subdir:
+    type: string
+  
+  os:
+    type: string
+  
+  arch:
+    type: string
+  
+  ext:
+    type: string
+  
+  no-extract:
+    type: boolean
+  
+  url-darwin:
+    type: string
+  
+  url-linux:
+    type: string
+  
+  url-win32:
+    type: string
+  
+  url-x64:
+    type: string
+  
+  url-darwin-x64:
+    type: string
+  
+  url-linux-x64:
+    type: string
+  
+  url-win32-x64:
+    type: string
+  
+  subdir-darwin:
+    type: string
+  
+  subdir-linux:
+    type: string
+  
+  subdir-win32:
+    type: string
+  
+  subdir-x64:
+    type: string
+  
+  subdir-darwin-x64:
```

Comment:
I've just merged #79 which also added `darwin-arm64` inputs. Can you add that here?

Reply:
> Can you add that here?

Sure, I added them now and of course also the new output.

But what about `subdir-darwin-arm64`?
That one is missing.
At least from your `actions.yml`.

---

> Ideally, I think, it would be possible to define types in a separate repository -- sort of like how you install separate @types/foo packages if using JS libraries with TypeScript -- and get the same result, without having to maintain them in-project. But I understand the appeal of this way too, so I have no issue merging it.

Actually, that *is* possible using https://github.com/typesafegithub/github-actions-typing-catalog.
But having the typings separated from the project always guarantees for outdated typings.
Having them within the project, the maintainer can right away maintain them properly, and you get exactly the types matching for that version, and also verify their consistency with the declared inputs and outputs via GHA workflow.

Also the difference to `@types/foo` is, that `@types/foo` is highly TypeScript specific just so that you can use the JS libs typed in tpyescript.

`[github-actions-typing](https://github.com/typesafegithub/github-actions-typing)` intentionally is made just independent typing information.
Also your "manual" users can benefit from having the types clearly and formally documented as well as any automated consumer like for example the Kotlin DSL binding generator can use them. :-)

Reply:
> But what about `subdir-darwin-arm64`?

Sound like a bug. The matrix of inputs that the design of this action requires is just hard to maintain.

Mind fixing that here (in `action.yml`) too?

Reply:
Sounds like you should automate its creation, or at least add some verification. :-)

Added.

Reply:
I don't disagree.